### ISSUE TITLE
Feature: detect overload

### ISSUE THREAD

**Issue by mranney** [tsenart/vegeta#126]
Sometimes a system under test cannot handle the expected load for unexpected reasons like the network is slow, there's other load on the system, etc. In this case, vegeta spawns more connections to attempt to increase throughput. However, this additional concurrency often has a negative impact on throughput, often resulting in both vegeta and the system under test to be very overloaded.

It would be very useful to measure whether adding more concurrency is actually increasing throughput, or more importantly, if it is decreasing throughput. I understand that this is a tricky thing to get 100% correct, because throughput drops could be unrelated to vegeta's actions. An option for enabling this behavior, even with false positives, would make it much safer to use on more systems.

**Comment by tsenart** [comment]
@mranney: Thanks for your feedback, I totally understand where you're coming from. However, vegeta was built and designed to generate **constant** load, not even as much load as it can...

> Vegeta is a versatile HTTP load testing tool built out of a need to drill HTTP services with a constant request rate. It can be used both as a command line utility and a library.

**Comment by mranney** [comment]
Yeah, I get that. The constant load design is why I started using it. :)

What I've found though is that it becomes risky to use in some environments where it would be very valuable because it can't back off when things go sideways. For example, we want to simulate peak load in production during off-peak times, but it's still production.

Anyway, I understand the desire to focus on your primary goal. Just thought I'd put that idea out there. If it ends up mattering enough to us, I may take a crack at fixing it myself.
